pkg/eval/hook.go

```diff
@@ -0,0 +1,40 @@
+package eval
+
+import (
+	"fmt"
+	"os"
+
+	"src.elv.sh/pkg/diag"
+	"src.elv.sh/pkg/eval/vals"
+)
+
+// CallHook runs all the functions in the list "hook", with "args". If
+// "cfg" is not specified, current eval's ports will be used.
```

Comment:
> current eval's ports will be used

This seems to contradict the implementation, which uses standard files to build ports. Evalers don't actually have associated ports (any more - they probably used to).

Also a TODO is warranted here, since none of the hooks should actually use the standard files:

 - Chdir hooks should use the frame from where `Evaler.Chdir` is called. This is the more problematic case as `cd` can easily be called with redirections.

 - Editor lifecycle should use the TTY passed to initialize it - this is less problematic because the editor is hooked to the stderr, but it makes testing impossible.